mandible/umm/ummg.py

```diff
@@ -0,0 +1,225 @@
+from dataclasses import asdict, dataclass
+from datetime import datetime, timezone
+from pathlib import Path
+from typing import List
+
+# ISO 8601 with Z
+UMM_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+
+
+def _remove_missing(ummg: dict) -> dict:
+    return {k: v for k, v in ummg.items() if v}
+
+
+@dataclass
+class FileMd(dict):
+    name: str
+    uri: str
+    s3uri: str
+    size: str
+    key: str
+    bucket: str
+
+
+@dataclass
+class CollectionRef:
+    ShortName: str
+    LongName: str
+
+
+@dataclass
+class ProductMd:
+    productCreationDateTime: datetime
+    productStartTime: datetime
+    productStopTime: datetime
+    mission: str
+
+
+@dataclass
+class ProductFilesMd:
+    productFile: FileMd
+    relatedFiles: List[FileMd]
+
+
+@dataclass
+class Meta:
+    CollectionRef: CollectionRef
+    ProductMd: ProductMd
+    ProductFilesMd: ProductFilesMd
+
+
+class UmmgBase():
+    def __init__(self, meta: Meta):
+        meta = asdict(meta)
```

Comment:
If you're requiring that the input be a dataclass anyway there's not much point in converting it to a dict internally right?

Reply:
Well I was having issues with the calls without doing this the meta object was not callable with the bracket syntax. The difference is maybe switching the whole class to use . Operator?

Reply:
Dataclasses don't use the bracket syntax, they use regular attribute access. i.e. `dataclass_obj.the_attribute`